snap/snapcraft.yaml

```diff
@@ -0,0 +1,256 @@
+      - libspnav-dev # optional
+      - libeigen3-dev
+      - libgts-bin
+      - libgts-dev
+      - libkdtree++-dev
+      - libmedc-dev
+      - libopencv-dev
+      - libproj-dev
+      - libx11-dev
+      - libxerces-c-dev
+      - libzipios++-dev
+      - swig
+      - python3-dev
+      - libcoin-dev
+      - libocct-data-exchange-dev
+      - libocct-draw-dev
+      - libocct-foundation-dev
+      - libocct-modeling-algorithms-dev
+      - libocct-modeling-data-dev
+      - libocct-ocaf-dev
+      - libocct-visualization-dev
+      - occt-draw
+      - libvtk7-dev
+      - libpyside2-dev
+      - libshiboken2-dev
+      - pybind11-dev
+    stage-packages:
+      - libaec0
+      - libboost-filesystem1.71.0
+      - libboost-program-options1.71.0
+      - libboost-python1.71.0
+      - libboost-regex1.71.0
+      - libboost-system1.71.0
+      - libboost-thread1.71.0
+      - libboost-date-time1.71.0 
+      - libhdf5-openmpi-103
+      - libhwloc15
+      - libilmbase24
+      - libjxr0
+      - libmedc11
+      - libmed11
+      - libopenexr24
+      - libopenmpi3
+      - libpython3.8
+      - libpython3.8-minimal
+      - libpython3.8-stdlib
+      - libraw19
+      - libspnav0
+      - libsz2
+      - libxerces-c3.2
+      - libzipios++0v5
+      - python3-tk # FEM
+      - python3-yaml # FEM
+      - python3-numpy
+      - python3-matplotlib
+      - python3-six
+      - python3-pyparsing
+      - python3-setuptools-scm
+      - python3-cycler
+      - python3-dateutil
+      - python3-git
+      - python3-ply # OpenSCAD
+      - python3-pivy
+      - python3-markdown # Addon manager
+      - python3-pyside2.qtcore
+      - python3-pyside2.qtgui
+      - python3-pyside2.qtsvg
+      - python3-pyside2.qtwidgets
+      - python3-pyside2.qtnetwork
+      - graphviz
+      - calculix-ccx # FEM
+      - libcoin80c
+      - libfreeimage3
+      - libocct-data-exchange-7.5
+      - libocct-foundation-7.5
+      - libocct-modeling-algorithms-7.5
+      - libocct-modeling-data-7.5
+      - libocct-ocaf-7.5
+      - libocct-visualization-7.5
+      - libtbb2
+      - libvtk7.1p
+      - elmerfem-csc # FEM
+      - openscad  # OpenSCAD
+    override-build: |
+      kde_sdk_dir="/snap/kde-frameworks-5-qt-5-15-3-core20-sdk/current"
+      mkdir -p /etc/xdg/qtchooser
+      cp "$kde_sdk_dir/etc/xdg/qtchooser/default.conf" "/etc/xdg/qtchooser/default.conf"
+      snapcraftctl build
+      sed -i -E \
+        "s|^Icon=(.*)|Icon=\${SNAP}/usr/share/icons/hicolor/scalable/apps/org.freecadweb.FreeCAD.svg|g" \
+        $SNAPCRAFT_PART_INSTALL/usr/share/applications/org.freecadweb.FreeCAD.desktop
+      cd $SNAPCRAFT_PART_SRC
+      iso_date=$(date -I)
+      git_hash=$(git rev-parse --short=8 HEAD)
+      version="$iso_date~$git_hash"
+      snapcraftctl set-version "$version"
+
+  assembly3:
+    plugin: dump
+    source: https://github.com/realthunder/FreeCAD_assembly3.git
+    organize:
+      "*": usr/Mod/Assembly3/
```

Comment:
This should be set to a specific tag/commit. Or is pulling HEAD always the only sensible thing to do? There's also the possibility of releasing assembly3 independently and have it plug into this package. But that's a little more involved, and it'd kinda duplicate addon manager functionality.

Reply:
Pulling HEAD is fine at the moment.